addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1725,12 +1730,32 @@ const RangeUserValueWidget = UnitUserValueWidget.extend({
     // Public
     //--------------------------------------------------------------------------
 
+    /**
+     * @override
+     */
+    getMethodsParams(methodName) {
+        const params = this._super(...arguments);
+        if (this.isCustomRange) {
+            params.possibleValues = this.customValues;
+        }
+        return params;
+    },
     /**
      * @override
      */
     async setValue(value, methodName) {
         await this._super(...arguments);
-        this.input.value = this._value;
+        this.input.value = this.isCustomRange ? this.customValues.indexOf(value) : this._value;
```

Comment:
Sorry, it's my bad, but there is in fact an even more generic way to handle this and this will be more consistent with the rest of the system. The values should not be defined on a `data-values` attribute but directly as part of the method data-... that is used with it.

If you check how checkboxes work, this is the same. You can define a value for the unchecked version and for the checked version, for each associated method. For example:
```xml
<we-checkbox data-select-class="hello|world"/>
```
is a checkbox which allows to switch between using the class "hello" and the class "world", see we are also using the `|` character and not `,` as I suggested by mistake earlier. This will probably not require too much changes... in fact, you will probably remove code: all widgets already loads those sets of values with this line:
```javascript
this._methodsParams.optionsPossibleValues[key] = dataValue.split(/\s*\|\s*/g);
```

So your `customValues` variable... is in fact `this._methodsParams.optionsPossibleValues[methodName]` and you won't even have to override `getMethodsParam`.

So adapt so that your final xml will be:
```xml
<we-range data-select-class="classA|classB|classC"/>
```
Thanks :+1:

Reply:
Yes this is the best way to implement the option, but for `we-range` we still need a way to set the min/max/step for the input on `start`, for now, it's all based on dataset.values.
I can add a workaround for that (set input values based on first data attribute with` | `separated string...) but it doesn't seem a good idea.
What do you think ?

Reply:
This is a good idea ;)

Reply:
Just have to count the number of possible values for the first method you found. If there are multiple methods, they are supposed to have the same number of possible values.